test/integration/node-specific/mongo_client.test.ts

```diff
@@ -0,0 +1,569 @@
+
+          // @ts-expect-error: exhaustAllowed is a protocol option
+          if (ns.toString() === 'admin.$cmd' && isHello(command) && options.exhaustAllowed) {
+            expect(options).property('socketTimeoutMS').to.equal(0);
+            stub.restore();
+            client.close(done);
+          }
+          stub.wrappedMethod.apply(this, args);
+        });
+      });
+    }
+  });
+
+  it('must respect a finite connectTimeoutMS for the streaming protocol', {
+    metadata: { requires: { topology: 'replicaset', mongodb: '>= 4.4' } },
+    test: function (done) {
+      const client = this.configuration.newClient({
+        connectTimeoutMS: 10,
+        heartbeatFrequencyMS: 500
+      });
+      client.connect(err => {
+        expect(err).to.not.exist;
+        const stub = sinon.stub(Connection.prototype, 'command').callsFake(function (...args) {
+          const ns = args[0];
+          const command = args[1];
+          const options = args[2] || {};
+
+          // @ts-expect-error: exhaustAllowed is a protocol option
+          if (ns.toString() === 'admin.$cmd' && isHello(command) && options.exhaustAllowed) {
+            expect(options).property('socketTimeoutMS').to.equal(510);
+            stub.restore();
+            client.close(done);
+          }
+          stub.wrappedMethod.apply(this, args);
+        });
+      });
+    }
+  });
+
+  context('explict #connect()', () => {
+    let client: MongoClient;
+    beforeEach(function () {
+      client = this.configuration.newClient(this.configuration.url(), {
+        monitorCommands: true
+      });
+    });
+
+    afterEach(async function () {
+      await client.close();
+    });
+
+    it(
+      'create topology and send ping when auth is enabled',
+      { requires: { auth: 'enabled' } },
+      async function () {
+        const commandToBeStarted = once(client, 'commandStarted');
+        await client.connect();
+        const [pingOnConnect] = await commandToBeStarted;
+        expect(pingOnConnect).to.have.property('commandName', 'ping');
+        expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+      }
+    );
+
+    it(
+      'permit operations to be run after connect is called',
+      { requires: { auth: 'enabled' } },
+      async function () {
+        const pingCommandToBeStarted = once(client, 'commandStarted');
+        await client.connect();
+        const [pingOnConnect] = await pingCommandToBeStarted;
+
+        const findCommandToBeStarted = once(client, 'commandStarted');
+        await client.db('test').collection('test').findOne();
+        const [findCommandStarted] = await findCommandToBeStarted;
+
+        expect(pingOnConnect).to.have.property('commandName', 'ping');
+        expect(findCommandStarted).to.have.property('commandName', 'find');
+        expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+      }
+    );
+
+    it(
+      'automatically connect upon first operation (find)',
```

Comment:
Any tests that are testing the implicit connect should probably be in a separate describe block - this block is labeled `explicit #connect`

Reply:
lmk what you think of the new organization